### DISCUSSION TITLE
Is there any windowname-like solution for tmux?

### DISCUSSION THREAD

**Discussion by ethagnawl** [jpalardy/vim-slime#348] [Q&A]
[This great blog post](https://blog.djha.skin/blog/developing-common-lisp-using-gnu-screen-rlwrap-and-vim/) keyed me in to a feature of GNU Screen and vim-slime that I didn't realize I was missing out on: sending output to a named GNU Screen "window". 

tmux doesn't _seem_ to offer anything like this out of the box, since -- AFAIK -- you can't specify an ID when creating a new pane. The obvious option I see is to detect the ID of the pane you want to send output to and remap your vim-slime config but that seems really clunky. Has anyone come up with a workaround that doesn't involve this sort of clunky and potentially error-prone scripting?

**Comment by jpalardy** [comment]
Hi @ethagnawl 

Though not _exactly_ what you asked, I'm getting great mileage from using `{last}`, but `man tmux` has other ones

```
     {last}            !    The last (previously active) pane
     {next}            +    The next pane by number
     {previous}        -    The previous pane by number
     {top}                  The top pane
     {bottom}               The bottom pane
     {left}                 The leftmost pane
     {right}                The rightmost pane
     {top-left}             The top-left pane
     {top-right}            The top-right pane
     {bottom-left}          The bottom-left pane
     {bottom-right}         The bottom-right pane
     {up-of}                The pane above the active pane
     {down-of}              The pane below the active pane
     {left-of}              The pane to the left of the active pane
     {right-of}             The pane to the right of the active pane
```

**Comment by ethagnawl** [comment]
Hey, @jpalardy. Thanks for getting back to me.

Coincidentally, I've been experimenting with those "special tokens" since I started thinking about solutions to this problem. I've been using `bottom-right`, which is better than a static index but I still run into issues semi-regularly where, for example, I've quickly split a pane and then forgotten to move it out of the "token" position and the output winds up in the wrong place. `last` is also not consistent enough for me.

I haven't dug into it yet but I am curious to know if there might be a way to lean on tmux's mark pane feature? I haven't used this feature at all and only came across it when I was reading the manual earlier this week. It seems like it _could potentially_ be a candidate for a solution, though.

**Comment by jpalardy** [comment]
I also used `TMUX_PANE` as a solid reference, in the past.

It depends how automated you want things to be. The numbers increment, so there's a way to force a layout with a target as a delta of your vim pane.

Or you could name and use the `-F` of `tmux list-session` …
